[JENKINS-50669] RingBufferLogHandler improvement

neaguandrei101 opened the pull request:
<!-- Comment: 
A great PR typically begins with the line below.
Replace XXXXX with the numeric part of the issue's id you created on JIRA.
Please note that if you want your changes backported into LTS, you will need to create a JIRA ticket for it. Read https://jenkins.io/download/lts/#backporting-process for more.
-->
See [JENKINS-50669](https://issues.jenkins-ci.org/browse/JENKINS-50669).

<!-- Comment: 
If the issue is not fully described in the ticket, add more information here (justification, pull request links, etc.).

 * We do not require JIRA issues for minor improvements.
 * Bugfixes should have a JIRA issue (backporting process).
 * Major new features should have a JIRA issue reference.
-->

### Proposed changelog entries

* Entry 1: Improved way to get the DefaultRingBufferSize
Old PR https://github.com/jenkinsci/jenkins/pull/4448. 
I created this because I had some conflicts with git and I had to redo my PRs.

<!-- Comment: 
The changelogs will be integrated by the core maintainers after the merge.  See the changelog examples here: https://jenkins.io/changelog/ -->

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] JIRA issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

<!-- For new API and extension points: Link to the reference implementation in open-source (or example in Javadoc) -->

### Desired reviewers

@mention

<!-- Comment:
If you need an accelerated review process by the community (e.g., for critical bugs), mention @jenkinsci/code-reviewers
-->

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [x] There are at least 2 approvals for the pull request and no outstanding requests for change
- [x] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [x] Proper changelog labels are set so that the changelog can be generated automatically
- [x] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a JIRA issue should exist and be labeled as `lts-candidate`

neaguandrei101 committed:
[JENKINS-50669] RingBufferLogHandler improvement

varyvol commented:
@neaguandrei101 could you please solve the merge conflict? Thanks!

neaguandrei101 committed:
Merge branch 'master' into RingBuffer

daniel-beck commented on core/src/main/java/hudson/WebAppMain.java:
```diff
@@ -112,13 +109,25 @@
+     * @return int This returns DEFAULT_RING_BUFFER_SIZE
+     * @see <a href="https://issues.jenkins-ci.org/browse/JENKINS-50669">JENKINS-50669</a>
+     */
+    public static int getDefaultRingBufferSize() {
```
Lacks either `@Restricted` or `@since`.

daniel-beck commented on core/src/main/java/hudson/util/RingBufferLogHandler.java:
```diff
@@ -54,6 +54,10 @@ public RingBufferLogHandler(int ringSize) {
         records = new LogRecord[ringSize];
     }
 
+    public static int getDefaultRingBufferSize() {
```
Lacks either `@Restricted` or `@since`.

rsandell replied:
Reading the ticket I don't think this should be restricted. The original intent was for plugins to be able to read the value. If it is restricted then that is not possible.

neaguandrei101 committed:
Javadoc fix and @Restricted added

neaguandrei101 committed:
Merge remote-tracking branch 'origin/RingBuffer' into RingBuffer

neaguandrei101 commented:
I have edited Javadoc and added version @since(2.255) and added @Restricted(NoExternalUse.class). If I understand correctly this makes the method only be able to be referenced inside the module.

MRamonLeon commented on core/src/main/java/hudson/util/RingBufferLogHandler.java:
```diff
@@ -54,6 +57,16 @@ public RingBufferLogHandler(int ringSize) {
+    /**
+     * @return int DEFAULT_RING_BUFFER_SIZE
+     * @see <a href="https://issues.jenkins-ci.org/browse/JENKINS-50669">JENKINS-50669</a>
+     * @since 2.255
```
I guess
```suggestion
     * @since TODO
```

rsandell replied:
```suggestion
     * @since TODO
```

MRamonLeon reviewed (approved):
LGTM, just a minor fix to do I guess

rsandell commented on core/src/main/java/hudson/WebAppMain.java:
```diff
@@ -112,13 +109,27 @@
+     * Written for the requirements of JENKINS-50669
+     * @return int This returns DEFAULT_RING_BUFFER_SIZE
+     * @see <a href="https://issues.jenkins-ci.org/browse/JENKINS-50669">JENKINS-50669</a>
+     * @since 2.255
```
```suggestion
     * @since TODO
```

neaguandrei101 committed:
[JENKINS-50669] @since TODO javadoc fix

neaguandrei101 committed:
[JENKINS-50669] Removed Restricted

varyvol commented:
@daniel-beck @rsandell it seems like your comments have been addressed. Could you please have a look? Thanks!

daniel-beck commented:
What's the point of this PR? With Evergreen being unmaintained, there's no use case for this. What am I missing?

@timja Can we revert this?

daniel-beck commented on core/src/main/java/hudson/util/RingBufferLogHandler.java:
```diff
@@ -23,6 +23,9 @@
 package hudson.util;
 
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
```
Unused imports
